### PR TITLE
Re-enable dds for flutter drive tests that use DevTools

### LINKED ISSUE #153057: Benchmarks stop gathering data after passing --no-dds to flutter drive.
The commit in question is https://github.com/flutter/flutter/commit/fc465348b9053eaca32c2a0ba15aa6319cb85f99

[An example is here](https://flutter-flutter-perf.skia.org/e/?begin=1722976891&end=1723063291&keys=Xdb96875782d9a2eb3f54ae24403ab66b&selected=commit%3D41907%26name%3D%252Carch%253Dintel%252Cbranch%253Dmaster%252Cconfig%253Ddefault%252Cdevice_type%253Dmokey%252Cdevice_version%253Dnone%252Chost_type%253Dlinux%252Csub_result%253DmaxRss%252Ctest%253Dcomplex_layout_scroll_perf__devtools_memory%252C&xbaroffset=41769). There may be others too but not sure.

Screenshot 2024-08-07 at 2 19 08 PM

### PR DESCRIPTION
**Pull request by zanderso**
Fixes https://github.com/flutter/flutter/issues/153057

### CODE REVIEW

**Review by jmagman — APPROVED** [review]

File: dev/devicelab/lib/framework/utils.dart [hunk 1]
```diff
@@ -478,9 +481,9 @@ List<String> _flutterCommandArgs(String command, List<String> options) {
         '5',
       ],
 
-    // DDS should be disabled for flutter drive in CI.
+    // DDS should generally be disabled for flutter drive in CI.
     // See https://github.com/flutter/flutter/issues/152684.
-    if (command == 'drive') '--no-dds',
+    if (command == 'drive' && !driveWithDds) '--no-dds',
```
Comment (jmagman) [inline dev/devicelab/lib/framework/utils.dart:486]: Need to pipe `driveWithDds` into this method though.

Comment (zanderso) [reply dev/devicelab/lib/framework/utils.dart:486]: I think the only spot it needs to go is where it's added in `perf_tests.dart` below.

Comment (zanderso) [reply dev/devicelab/lib/framework/utils.dart:486]: Oh derp. Reading fail.

Comment (zanderso) [reply dev/devicelab/lib/framework/utils.dart:487]: Fixed.

**Review by chinmaygarde — APPROVED** [review]

**Pull request merged**